gammapy/irf/core.py

```diff
@@ -1,17 +1,284 @@
+
+    @lazyproperty
+    def _interpolate(self):
+        points = [a.center for a in self.axes]
+        points_scale = tuple([a.interp for a in self.axes])
+        return ScaledRegularGridInterpolator(
+            points, self.quantity, points_scale=points_scale, **self.default_interp_kwargs
+        )
+
+    @property
+    def quantity(self):
+        """`~astropy.units.Quantity`"""
+        return u.Quantity(self.data, unit=self.unit, copy=False)
+
+    @quantity.setter
+    def quantity(self, val):
+        val = u.Quantity(val, copy=False)
+        self.data = val.value
+        self.unit = val.unit
+
+    @property
+    def axes(self):
+        """`MapAxes`"""
+        return self._axes
+
+    def __str__(self):
+        str_ = f"{self.__class__.__name__}\n"
+        str_ += "-" * len(self.__class__.__name__) + "\n\n"
+        str_ += f"\taxes  : {self.axes.names}\n"
+        str_ += f"\tshape : {self.data.shape}\n"
+        str_ += f"\tndim  : {len(self.axes)}\n"
+        str_ += f"\tunit  : {self.unit}\n"
+        str_ += f"\tdtype : {self.data.dtype}\n"
+        return str_.expandtabs(tabsize=2)
+
+    def evaluate(self, method=None, **kwargs):
+        """Evaluate IRF
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Coordinates at which to evaluate the IRF
+        method : str {'linear', 'nearest'}, optional
+            Interpolation method
+
+        Returns
+        -------
+        array : `~astropy.units.Quantity`
+            Interpolated values
+        """
+        # TODO: change to coord dict?
+        non_valid_axis = set(kwargs).difference(self.axes.names)
+        if non_valid_axis:
+            raise ValueError(
+                f"Not a valid coordinate axis {non_valid_axis}"
+                f" Choose from: {self.axes.names}"
+            )
+
+        coords_default = self.axes.get_coord()
+
+        for key, value in kwargs.items():
+            coord = kwargs.get(key, value)
+            if coord is not None:
+                coords_default[key] = u.Quantity(coord, copy=False)
+
+        return self._interpolate(coords_default.values(), method=method)
+
+    # TODO: define a proper integration method
+    def integrate_energy(self, method="linear", **kwargs):
+        """Integrate in a given energy band.
+
+        This method uses log-log trapezoidal integration.
+
+        Parameters
+        ----------
+        **kwargs : dict
+            Coordinates at which to evaluate the IRF
+        method : {'linear', 'nearest'}, optional
+            Interpolation method
 
+        Returns
+        -------
+        array : `~astropy.units.Quantity`
+            Returns 2D array with axes offset
+        """
+        axis = self.axes.index("energy")
+        data = self.evaluate(**kwargs, method=method)
+        energy = kwargs["energy"]
+        return trapz_loglog(data, energy, axis=axis)
+
+    # TODO: define a proper integration method
+    @lazyproperty
```

Comment:
Same comment here. This is applicable only to PSF and would rather fit on a general PSF base class.